.github/workflows/build-test.yaml

```diff
@@ -206,36 +206,91 @@ jobs:
       - name: Run K3s and master in it
         run: tests/run-master-in-k3s.sh local/freeipa-server:${{ matrix.os }} freeipa-server-${{ matrix.os }}.tar
 
+  push-after-success-precondition:
+    name: Check pipeline settings
+    needs: [ test-docker, test-podman, test-rootless-podman, test-upgrade, test-k3s ]
+    runs-on: ubuntu-20.04
+    env:
+      DOCKER_TARGET_LIST: "${{ secrets.DOCKER_TARGET_LIST }}"
+      DOCKER_CREDENTIALS_FILE: "${{ secrets.DOCKER_CREDENTIALS_FILE }}"
+    steps:
+      - name: Check secrets
+        id: check
+        run: |
+          [ -e "${HOME}/.docker" ] || mkdir -p "${HOME}/.docker"
+          cat > "${HOME}/.docker/config.json" <<EOF
+          ${DOCKER_CREDENTIALS_FILE}
+          EOF
+          [ "${DOCKER_TARGET_LIST}" == "" ] && echo "Target list is empty" && echo "::set-output name=are-valid-credentials::false" && exit 0
+          IFS=":" read -a TARGET_LIST <<< "${DOCKER_TARGET_LIST}"
+          for base_name in "${TARGET_LIST[@]}"; do
+            docker login ${base_name%%/*} < /dev/null || {
+              echo "Can not log in to '${base_name%%/*}'' registry"
+              echo "::set-output name=are-valid-credentials::false"
+              exit 0
+            }
+          done
+          echo "::set-output name=are-valid-credentials::true"
+    outputs:
+      are-valid-credentials: ${{ steps.check.outputs.are-valid-credentials }}
+
   push-after-success:
-    name: Push images to Docker Hub
+    name: Push images to Registry
     runs-on: ubuntu-20.04
-    needs: [ test-docker, test-podman, test-rootless-podman, test-upgrade, test-k3s ]
-    if: github.event_name != 'pull_request' && github.repository == 'freeipa/freeipa-container' && github.ref == 'refs/heads/master'
+    needs: [ push-after-success-precondition ]
+    if: needs.push-after-success-precondition.outputs.are-valid-credentials == 'true' && github.event_name != 'pull_request' && ( ( github.repository == 'freeipa/freeipa-container' && github.ref == 'refs/heads/master' ) || github.repository != 'freeipa/freeipa-container' )
```

Comment:
I still have concerns about this. We need to prevent the pushes on fork's random branches. So either the check for `master` needs to be there for all repos (even the forks), or we need to parametrize the list of branches to push on with a secret as well.
In fact, one possibility would be to have that `PUSH_TARGET_LIST` to be lines in the form `<branch>:<repo>:<optional list of tags to push>`.

Reply:
Got it! I have restored this part. Thanks!